plugins/index.ts

```diff
@@ -116,47 +137,56 @@ const deployOrUpgradeProxy = async <
       contract.address
     );
   } else {
-    hre.trace(
-      'Found existing proxy at:',
-      maybeProxyAddress,
-      'attempting to upgrade instance',
-      contractName
-    );
-    const existingImplementationAddress =
-      await hre.upgrades.erc1967.getImplementationAddress(maybeProxyAddress);
-    hre.trace('Existing implementation at:', existingImplementationAddress);
-    const fireblocksSigner = signer as FireblocksSigner;
-    if (typeof fireblocksSigner.setNextTransactionMemo === 'function') {
-      fireblocksSigner.setNextTransactionMemo(
-        `Upgrade contract instance for ${contractName}`
-      );
-    }
-    const deployment = await hre.deployments.get(contractName);
-    const artifact = await hre.deployments.getArtifact(contractName);
-    if (deployment.bytecode !== artifact.bytecode) {
-      contract = await hre.upgrades.upgradeProxy<TContract>(
+    try {
+      hre.trace(
+        'Found existing proxy at:',
         maybeProxyAddress,
-        contractFactory,
-        { ...options }
+        'attempting to upgrade instance',
+        contractName
       );
-      const newImplementationAddress =
-        await hre.upgrades.erc1967.getImplementationAddress(maybeProxyAddress);
-      if (existingImplementationAddress === newImplementationAddress) {
-        hre.trace('Implementation unchanged');
+      const existingImplementationAddress =
+        await hre.upgrades.erc1967.getImplementationAddress(maybeProxyAddress!);
+      hre.trace('Existing implementation at:', existingImplementationAddress);
+      const fireblocksSigner = signer as any as FireblocksSigner;
+      if (typeof fireblocksSigner.setNextTransactionMemo === 'function') {
+        fireblocksSigner.setNextTransactionMemo(
+          `Upgrade contract instance for ${contractName}`
+        );
+      }
+      const deployment = await hre.deployments.get(contractName);
+      const artifact = await hre.deployments.getArtifact(contractName);
+      if (deployment.bytecode !== artifact.bytecode) {
+        contract = await hre.upgrades.upgradeProxy<TContract>(
+          maybeProxyAddress!,
```

Comment:
Would be nice to maybe include this in the if statement above  (IE - `if (deployment.bytecode !== artifact.bytecode && maybeProxyAddress !== undefined)` so we don't have to assert that it's defined here and everywhere else it's used

Reply:
so the only reason this is showing up as a diff in this PR is because of some large-scale block rearrangements into try/catch statements.  I hesitate to adjust this logic now because I'd like the contents of this file to reflect what they were at the time they were used to successfully create a deployment.  I think if we want to update this in the future we should do so in a separate PR and ideally with a deployment of some sort to verify that they work?